_data/rules.yml

```diff
@@ -998,6 +998,66 @@ groups:
           - name: instaclustr/cassandra-exporter
             doc_url: https://github.com/instaclustr/cassandra-exporter
             rules:
+              - name: 'Cassandra Node is unavailable
+                description: 'Cassandra Node is unavailable - {{ $labels.cassandra_cluster }} {{ $labels.exported_endpoint }}'
+                query: 'sum(min_over_time(cassandra_endpoint_active{cassandra_cluster=~".*"}[1m])) by (cassandra_cluster,instance,exported_endpoint) < 1'
+                for: 2m
+                severity: critical
+              - name: 'Cassandra many compaction tasks are pending'
+                description: 'Many Cassandra compaction tasks are pending - {{ $labels.cassandra_cluster }}'
+                query: 'avg_over_time(cassandra_table_estimated_pending_compactions{cassandra_cluster=~".*"}[1m]) > 100'
+                for: 2m
+                severity: warning
+              - name: 'Cassandra commitlog pending tasks'
+                description: 'Cassandra commitlog pending tasks - {{ $labels.cassandra_cluster }}'
+                query: 'avg_over_time(cassandra_commit_log_pending_tasks{cassandra_cluster=~".*"}[1m]) > 15'
+                for: 2m
+                severity: warning
+              - name: 'Cassandra compaction executor blocked tasks'
+                description: 'Some Cassandra compaction executor tasks are blocked - {{ $labels.cassandra_cluster }}'
+                query: 'avg(avg_over_time(cassandra_thread_pool_blocked_tasks{cassandra_cluster=~".*",pool="CompactionExecutor"}[1m])) by (cassandra_cluster,instance) > 15'
```

Comment:
Is it necessary to `avg()`? Since it is an issue local to a node (maybe I'm wrong!).

Reply:
yes it is per node. so you suggest using this instead?
```
query: 'avg_over_time(cassandra_thread_pool_blocked_tasks{pool="CompactionExecutor"}) > 15`
```

Reply:
Done for both `CompactionExecutor` and `MemtableFlushWriter`.